clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/Metadata.scala

```diff
@@ -100,4 +103,43 @@ object Metadata {
     val srcBase = srcName.take(srcName.toLowerCase.lastIndexOf(extension))
     destination.resolve(s"${newBasename.getOrElse(srcBase)}$extension")
   }
+
+  /**
+    * Given one of our `Metadata` classes, extract out all fields that
+    * store path-related information into a map from 'fieldName' -> 'path'.
+    *
+    * Used to build a generic before-and-after move comparison to determine
+    * which paths in a metadata object will actually be affected by the move.
+    */
+  def extractPaths[M <: Metadata[M]: ClassTag](metadata: M): Map[String, URI] = {
```

Comment:
I think this whole thing could get ripped out & replaced by `JsonDiff`... will play with it